src/duckula/avro/schema.clj

```diff
@@ -34,27 +37,37 @@
     ;; Complex Types
     Schema$Type/RECORD
     (apply hash-map (mapcat (fn [key val]
-                              [(keyword key) val])
-                            (map #(.name %) (.getFields avro-schema))
-                            (map #(->map (.schema %)) (.getFields avro-schema))))
+                              (let [k (if mangle-names?
+                                        (-> key (str/replace \_ \-) keyword)
+                                        (keyword key))]
+                                [k val]))
+                            (map #(.name ^Schema$Field %) (.getFields ^Schema avro-schema))
+                            (map #(->map {:avro-schema (.schema ^Schema$Field %)
+                                          :mangle-names? mangle-names?})
+                                 (.getFields ^Schema avro-schema))))
 
     Schema$Type/ENUM
-    (apply s/enum (.getEnumSymbols avro-schema))
+    (apply s/enum (map (fn [enum]
+                         (if mangle-names?
+                           (str/replace enum \_ \-)
+                           enum))
```

Comment:
Since is also used above, should this be a function?

Reply:
Definitely!